core/src/main/scala/kafka/common/InterBrokerSendThread.scala

```diff
@@ -32,17 +33,18 @@ import scala.jdk.CollectionConverters._
 /**
  *  Class for inter-broker send thread that utilize a non-blocking network client.
  */
-abstract class InterBrokerSendThread(name: String,
-                                     networkClient: KafkaClient,
-                                     time: Time,
-                                     isInterruptible: Boolean = true)
-  extends ShutdownableThread(name, isInterruptible) {
-
-  def generateRequests(): Iterable[RequestAndCompletionHandler]
-  def requestTimeoutMs: Int
+class InterBrokerSendThread(
+  name: String,
+  networkClient: KafkaClient,
+  requestTimeoutMs: Int,
+  time: Time,
+  isInterruptible: Boolean = true
+) extends ShutdownableThread(name, isInterruptible) {
+
+  private val inboundQueue = new ConcurrentLinkedQueue[RequestAndCompletionHandler]()
```

Comment:
The `BrokerToControllerChannelManager` is probably the only case where this might be a concern since requests can be forwarded form the client. We wouldn't want to block the request handler if the queue gets too large, but we might want to start failing inbound requests. I'd suggest we create a separate JIRA to think about this. Sound fair?

Reply:
Yea, sounds good 👍